src/backend/utils/adt/like_match.c

```diff
@@ -198,6 +204,66 @@ MatchText(const char *t, int tlen, const char *p, int plen,
 			NextByte(p, plen);
 			continue;
 		}
+		else if (*p == '[' && sql_dialect == SQL_DIALECT_TSQL)
```

Comment:
Where are the test cases?

Reply:
the corresponding test cases is 
b like '%[abc]'